lib/initial-setup/WP_Auth0_InitialSetup_Consent.php

```diff
@@ -128,19 +144,21 @@ public function consent_callback( $name ) {
 		$connection_exists     = false;
 		$connection_pwd_policy = null;
 
-		$connections = WP_Auth0_Api_Client::search_connection( $domain, $app_token );
+		$connections = WP_Auth0_Api_Client::search_connection( $domain, $app_token, 'auth0' );
 
-		foreach ( $connections as $connection ) {
+		if ( $should_create_and_update_connection && ! empty( $connections ) && is_array( $connections ) ) {
```

Comment:
Logic here is still not great but did not want to completely refactor this in a patch release. Main thing here is checking whether we have an array and avoiding the loop if we're not using the values later.

Reply:
I understand but looking for the minimum number of changes here. This is just one of many things that should be addressed in this method.